crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs

```diff
@@ -0,0 +1,327 @@
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let DocstringExtraneousException { ids } = self;
+        format!("{} not explicitly raised.", ids.join(", "))
+    }
+}
+
+// Parse docstring
+#[derive(Debug)]
+struct DocstringEntries {
+    raised_exceptions: Vec<String>,
+    raised_exceptions_range: Option<TextRange>,
+}
+
+impl DocstringEntries {
+    fn new(sections: &SectionContexts, style: SectionStyle) -> Self {
+        let mut raised_exceptions: Vec<String> = Vec::new();
+        let mut raised_exceptions_range = None;
+
+        for section in sections.iter() {
+            if section.kind() == SectionKind::Raises {
+                raised_exceptions = parse_entries(section.following_lines_str(), style);
+                raised_exceptions_range = Some(section.range());
+            }
+        }
+
+        Self {
+            raised_exceptions,
+            raised_exceptions_range,
+        }
+    }
+}
+
+fn parse_entries(content: &str, style: SectionStyle) -> Vec<String> {
+    match style {
+        SectionStyle::Google => parse_entries_google(content),
+        SectionStyle::Numpy => parse_entries_numpy(content),
+    }
+}
+
+fn parse_entries_google(content: &str) -> Vec<String> {
+    let mut entries: Vec<String> = Vec::new();
+    for potential in content.split('\n') {
+        let Some(colon_idx) = potential.find(':') else {
+            continue;
+        };
+        let entry = potential[..colon_idx].trim().to_string();
+        entries.push(entry);
+    }
+    entries
+}
+
+fn parse_entries_numpy(content: &str) -> Vec<String> {
+    let mut entries: Vec<String> = Vec::new();
+    let mut split = content.split('\n');
+    let Some(dashes) = split.next() else {
+        return entries;
+    };
+    let indentation = dashes.len() - dashes.trim_start().len();
+    for potential in split {
+        if let Some(first_char) = potential.chars().nth(indentation) {
+            if !first_char.is_whitespace() {
+                let entry = potential[indentation..].trim().to_string();
+                entries.push(entry);
+            }
+        }
+    }
+    entries
+}
+
+// Parse body
+#[derive(Debug)]
+struct Entry {
+    id: String,
+    range: TextRange,
+}
+
+struct BodyEntries {
+    raised_exceptions: Vec<Entry>,
+}
+
+struct BodyVisitor<'a> {
+    raised_exceptions: Vec<Entry>,
+    semantic: &'a SemanticModel<'a>,
+}
+
+impl<'a> BodyVisitor<'a> {
+    fn new(semantic: &'a SemanticModel<'a>) -> Self {
+        Self {
+            raised_exceptions: Vec::new(),
+            semantic,
+        }
+    }
+
+    fn finish(self) -> BodyEntries {
+        BodyEntries {
+            raised_exceptions: self.raised_exceptions,
+        }
+    }
+}
+
+impl Visitor<'_> for BodyVisitor<'_> {
+    fn visit_stmt(&mut self, stmt: &Stmt) {
+        if let Stmt::Raise(ast::StmtRaise { exc: Some(exc), .. }) = stmt {
+            match exc.as_ref() {
+                Expr::Name(ast::ExprName { id, range, .. }) => {
```

Comment:
Per my comment above, I wonder if we need to match on attributes here too. What do you think?